test/test_nn.py

```diff
@@ -16646,6 +16646,23 @@ def test_conv2d_no_grad(self, device, dtype):
                 output = m(input)
                 self.assertEqual(output, output_ng, rtol=1e-2, atol=1e-5)
 
+    @onlyCUDA
+    @skipCUDAIfNoCudnn
+    @dtypes(torch.float, torch.double, torch.float16)
+    def test_cudnn_convolution_relu(self, device, dtype):
+        for batch in [1, 2, 3]:
```

Comment:
nit: 4 nested levels is too much, also, use tuples instead of lists:
```
for batch, groups, kernel_size, memory_format in product((1, 2, 3), (1, 2, 4), ((1, 1), (3, 3)), (torch.channels_last, torch_contiguous_format)):
```